keps/sig-storage/1710-selinux-relabeling/README.md

```diff
@@ -131,127 +146,189 @@ For CSI, kubelet currently uses following heuristics:
 
 ### Goals
 
-* If possible, mount volumes with the correct SELinux context using `-o context=XYZ` mount option and avoid recursive change of all files on the volume.
-
+* Mount volumes with the correct SELinux label using `-o context=XYZ` mount option and avoid recursive change of all files on the volume.
+  * Do it _by default_ for volumes that can't be shared among Pods, i.e. ReadWriteOncePod PVs.
+  * Do it as opt-in for all other volumes, both in-line in Pods or PersistentVolumes.
+    As explained above, mixing privileged and unprivileged Pods sharing the same volume is a use case that works today. It can't work with `-o context` mount, see [example below](#conflicts-with-other-pods) with Privileged pods.  
+    
 ### Non-Goals
 
 * Change container runtimes / CRI.
 
 ## Proposal
 
-* When kubelet *knows* SELinux context of a pod (i.e. at least `Pod.Spec.SecurityContext.SELinuxOptions.Level` is set) AND kubelet *knows* that a volume supports mounting with `-o context`:
-  * kubelet passes `-o context=XYZ` to `MountDevice()` and `SetUp()` calls of the volume.
-    * A volume plugin / CSI driver will get these as regular mount options and use them to mount the volume.
-    * Especially for shared filesystems, it's task of a CSI driver to:
-      * Detect `context` mount option and add any other mount option that allow mount options that's needed to allow mounting one volume with multiple SELinux contexts on the same node.
-        E.g. add `nosharesock` to `mount -t cifs -o context=XYZ`.
-      * Either not support `NodeStageVolume` CSI call or use different VolumeHandle for each PV, even though they lead to the same volume in the storage backend.
-        * This is because kubelet / VolumeManager can track only one global volume mount (created by `NodeStageVolume`) and that mount can have only one SELinux context.
-          Volumes with different VolumeHandles have different global mounts and therefore can have different SELinux contexts.
-  * kubelet does not pass any special SELinux option to the container runtime (explicitly, no `:Z`).
-    Files on the volume already have the right SELinux context, no recursive relabeling happens there.
-  * When a `PodSecurityContext` specifies incomplete SELinux label (i.e. omits `SELinuxOptions.User`, `.Role` or `.Type`), kubelet fills the blanks from the system defaults provided by [ContainerLabels() from go-selinux bindings](https://github.com/opencontainers/selinux/blob/621ca21a5218df44259bf5d7a6ee70d720a661d5/go-selinux/selinux_linux.go#L770).
+Introduce a new PodSpec field `Spec.SecurityContext.SELinuxChangePolicy` with the following values:
+* `UseMountOptionForReadWriteOncePod`: mount RWOP volumes with `-o context` (see other conditions below), all other volumes are relabeled recursively by the container runtime.
```

Comment:
Another option I was thinking is - `UseMountOptionsSafely`, it is perhaps as much big as current choice, but again this name is going to be just picking "okay" option among worst. :(

Reply:
Since the new default is opt-out, we don't need `UseMountOptionsSafely`